src/segmented_buffer.rs

```diff
@@ -311,22 +450,6 @@ impl<F> std::io::Write for SegmentedPoolBuf<F, BytesMut, AllocBytesMutFn> {
                         Ok(mut new_buf) => {
                             // clear the BytesMut
                             new_buf.clear();
```

Comment:
From security perspective, the pool should be responsible to return cleared buffers.

Reply:
I added a trait so that the buf pool will do this over at https://github.com/logdna/async-buf-pool-rs/tree/convenience-impls, this makes the buf pool a lot less general, but we're the only ones using it, so I think it's worthwhile

Reply:
Cool!